lib/SILGen/SILGenExpr.cpp

```diff
@@ -3782,7 +3783,69 @@ visitMagicIdentifierLiteralExpr(MagicIdentifierLiteralExpr *E, SGFContext C) {
   llvm_unreachable("Unhandled MagicIdentifierLiteralExpr in switch.");
 }
 
-RValue RValueEmitter::visitCollectionExpr(CollectionExpr *E, SGFContext C) {
+RValue RValueEmitter::visitArrayExpr(ArrayExpr *E, SGFContext C) {
```

Comment:
As a meta comment. This needs SILGen tests that exercise this code path. I noticed that you didn't add any. Is that b/c this is a NFC commit? If not, it is important that you add SILGen tests.

Also can you be consistent about variable capitalization.